### PR TITLE
Close the underlying ws connections after 50 millseconds

### PR DESCRIPTION
**Pull request by mstoykov**
Previous to this the event loop can be blocked on slow operation
-read/write, and not actually see that the context has been canceled.

With this change the underlying connection will be closed 5 milliseconds
after the context is which circumvents the whole event loop. This also
means that the other side will likely not get CloseMessage and that
`close` or `error` handlers will not be called. The latter of which
would've likely not ran either way given that the goja VM would've also
probably be interrupted.

### CODE REVIEW

File: js/modules/k6/ws/ws_test.go [hunk 1]
```diff
@@ -1317,3 +1320,89 @@ func TestCookieJar(t *testing.T) {
...
+			t.Fatalf("/ws-echo-someheader cannot upgrade request: %v", err)
+		}
+		_, r, err := conn.NextReader() // this is slightly faster than `conn.ReadMessage`
+		if err != nil {
+			ch <- err
+			close(ch)
+			return
+		}
+		n, _ := io.Copy(io.Discard, r)
+		if n == 9000 { // this is exactly how big it should be
```
Comment (olegbespalov) [inline js/modules/k6/ws/ws_test.go:1341]: This condition looks a bit brittle. Should it be exactly 9000?

File: js/modules/k6/ws/ws.go [hunk 2]
```diff
@@ -283,12 +283,39 @@ func (mi *WS) Connect(url string, args ...goja.Value) (*WSHTTPResponse, error) {
...
+	}()
+
+	go func() {
+		// Wait for the context to be over and close the underlying connection directly
+		// This will *not* send CloseMessage or trigger any `close` events but prevents a slow event handler to take forever
+		select {
+		case <-ctx.Done():
+			t := time.NewTimer(time.Millisecond * 50)
+			select {
+			case <-t.C: // wait 5 milliseconds in case the event loop below isn't blocked
```
Comment (olegbespalov) [inline js/modules/k6/ws/ws.go:310]: Do we need this comment? Is it five milliseconds?

**Review by codebien — COMMENTED** [review]
Ping me when this is ready for review

**Comment by mstoykov** [comment]
This is a hacky fix for a very rare situation. 

And given that we will likely replace the current API with something else, for example https://github.com/grafana/k6/issues/2425 it doesn't make much sense to persue this

**Pull request closed without merging**